### PR TITLE
Add nunito sans font

### LINKED ISSUE #20828: Add the Nunito Sans typeface to Storybook
Storybook was designed to use Nunito Sans. It was never added even if it was listed in the font stack. We will add the Storybook typeface to Storybook.

### PR DESCRIPTION
**Pull request by ndelangen**
Resolve: https://github.com/storybookjs/storybook/issues/20828
Resolve: https://github.com/storybookjs/storybook/issues/16278

## What I did

- I add a few font files
- I add these files to be hosted / copied by storybook via a preset config
- I add references to these font files in the manager builder's template

## How to test

- start a storybook in dev-mode
- open the browser
- expect the network tab to show you the browser loading 2 font files in the manager, followed by a css file.
- expect the font defined in that css files to be used in the manager

repeat the above steps but for a build storybook. (you can use a chromatic deployed storybook for this)

### CODE REVIEW

File: code/lib/core-server/src/presets/common-preset.ts [hunk 1]
```diff
@@ -13,32 +13,41 @@ import type {
...
 } from '@storybook/types';
 import { loadCsf } from '@storybook/csf-tools';
-import { join } from 'path';
+import { dirname, join } from 'path';
 import { dedent } from 'ts-dedent';
 import { parseStaticDir } from '../utils/server-statics';
 
 const defaultFavicon = require.resolve('@storybook/core-server/public/favicon.svg');
 
+export const staticDirs: PresetPropertyFn<'staticDirs'> = async (values = []) => [
```
Comment (yannbf) [inline code/lib/core-server/src/presets/common-preset.ts:25]: 😱 I had no idea this could be used as preset!

**Comment by yannbf** [comment]
Interesting (and understandable) that Chromatic didn't pick up visual changes, but here are a few I noticed:

**Comment by ndelangen** [comment]
I was instructed to use "Regular" and "Black" variation, thus that's the fonts-files I added.

@MichaelArestad can you please comment?

**Comment by MichaelArestad** [comment]
I updated the typefaces to include Regular, Italic, Bold, and Bold Italic. I removed black and updated the few places the 900 weight was used.

1. Is there concern in removing the `black` option from the theming package?
2. The italic weight would only be used on docs pages. Is it possible to conditionally load/preload the italic typefaces only for docs pages? Or perhaps load them after everything else on non-docs pages?

**Review by ndelangen — COMMENTED** [review]
nice!

**Comment by ndelangen** [comment]
> Is there concern in removing the black option from the theming package?

No, I have no concerns about that at all.

---

> The italic weight would only be used on docs pages. Is it possible to conditionally load/preload the italic typefaces only for docs pages? Or perhaps load them after everything else on non-docs pages?

This PR doesn't impact the preview (where docs renders) at all.
If the user doesn't have nunito sans installed, they'll still see the fallback font in the preview, even after merging this PR.

Am I correct in assuming you want me to apply this font-loading change to for the preview as well? @MichaelArestad

**Comment by MichaelArestad** [comment]
> Am I correct in assuming you want me to apply this font-loading change to for the preview as well? @MichaelArestad

@ndelangen Yes, please. The only fonts the manager use are regular and bold. Only the docs pages would utilize the italic variants.

**Review by valentinpalkovic — APPROVED** [review]

**Comment by valentinpalkovic** [comment]
@ndelangen Thank you, Norbert, for taking care of the requested changes. Should we merge? :)

**Pull request merged**